data_processing_pipeline/2_process/src/process_gages_utils.R

```diff
@@ -41,6 +41,33 @@ process_site_info <- function(target_name, active_gages_data_file, file.out = FA
   }
 }
 
+#' create state-specific year counts of number of active gages
+#' writes a file for a data.frame that has three columns: state, year, and n_gages
+process_state_counts <- function(file_out, summary_fl){
+  
+  gage_summary_data <- readRDS(summary_fl)
+  
+  gages_info <- gage_summary_data %>% 
+    pull(site) %>% unique() %>% 
+    dataRetrieval::readNWISsite() %>% rename(STATE = state_cd) %>% 
+    full_join(dataRetrieval::stateCd, by = "STATE") %>%
+    select(site_no, state = STUSAB)
+  
+  if (any(is.na(gages_info$state))){
+    stop("one or more sites didn't have a state code that joined")
+  }
+  
+  # now re-join with the new `state` column
+  gage_summary_data %>% rename(site_no = site) %>% inner_join(gages_info, by = "site_no") %>% 
+    unnest(which_years_active) %>% rename(year = which_years_active) %>% select(site_no, year, state) %>% 
+    group_by(state, year) %>% 
+    summarize(n_gages = length(year)) %>% 
```

Comment:
Does this need to be `length(site_no)`?

Reply:
it is the same number in this case - just the `nrow()` of the data.frame filtered per state+year

Do you want me to change it? I think I will for readability's sake

Reply:
I guess that's true since we aren't doing `length(unique(year))`. I would vote to change for readability just because it makes it seem like you would just get 1 every time since that is a grouping variable.